src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -110,14 +115,44 @@ private IntPtr AllocException(Exception ex)
         [DllImport(JitSupportLibrary)]
         private extern static char* GetExceptionMessage(IntPtr obj);
 
-        public static void Startup()
+
+        public static void Startup(TargetDetails target, string jitPathOverride)
+        {
+            Debug.Assert(s_jitLibrary == IntPtr.Zero);
+
+            if (!string.IsNullOrEmpty(jitPathOverride))
+            {
+                s_jitLibrary = NativeLibrary.Load(jitPathOverride);
+            }
+            else
+            {
+                string jitLibraryName = JitLibrary + '-' + GetTargetSpec(target);
+                s_jitLibrary = NativeLibrary.Load(jitLibraryName, typeof(CorInfoImpl).Assembly, searchPath: null);
+            }
+
+            s_jitStartup = Marshal.GetDelegateForFunctionPointer<JitStartupDelegate>(NativeLibrary.GetExport(s_jitLibrary, "jitStartup"));
+            s_getJit = Marshal.GetDelegateForFunctionPointer<GetJitDelegate>(NativeLibrary.GetExport(s_jitLibrary, "getJit"));
```

Comment:
What was wrong with the existed `NativeLibrary.SetDllImportResolver` approach?

Reply:
As JanK pointed out, with the changed approach to locating the JIT dll it no longer makes much sense - we either consume the stock clrjit next to Crossgen2 as identified by the targeting OS / CPU architecture, or explicitly load the custom one as specified in the JitPath command-line argument.

Reply:
> we either consume the stock clrjit next to Crossgen2 as identified by the targeting OS / CPU architecture, or explicitly load the custom one as specified in the JitPath command-line argument

And that logic could easily be placed in the `NativeLibrary.SetDllImportResolver` callback.  Am I missing anything?

Reply:
+1 - The SetDllImportResolver is cleaner and a lot fewer lines of code than juggling delegates. CorInfoImpl no longer needs to worry about paths. Are there any drawbacks I'm missing?

Reply:
(I think Jan's comment was that since you're juggling delegates, you don't need SetDllImportResolver, but Anton and I are questioning the juggling of delegates in the first place.)

Reply:
I have modified the change based on your feedback in 25th commit by basically reverting all remaining changes to CorInfoImpl and just handling the clrjit name translation logic in SetDllImportResolver.